webpack/assets/javascripts/react_app/components/HostDetails/Tabs/Details/Cards/Virtualization/index.js

```diff
@@ -0,0 +1,82 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import {
+  DescriptionList,
+  DescriptionListTerm,
+  DescriptionListGroup,
+  DescriptionListDescription,
+} from '@patternfly/react-core';
+import { translate as __ } from '../../../../../../common/I18n';
+import { foremanUrl } from '../../../../../../common/helpers';
+import { useAPI } from '../../../../../../common/hooks/API/APIHooks';
+import CardTemplate from '../../../../Templates/CardItem/CardTemplate';
+import SkeletonLoader from '../../../../../common/SkeletonLoader';
+import DefaultLoaderEmptyState from '../../../../DetailsCard/DefaultLoaderEmptyState';
+import { STATUS } from '../../../../../../constants';
+import VirtVmware from './VirtVmware';
+import VirtOvirt from './VirtOvirt';
+import VirtLibvirt from './VirtLibvirt';
+import VirtEc2 from './VirtEc2';
+import VirtGce from './VirtGce';
+import VirtOpenstack from './VirtOpenstack';
+
+const VirtualizationCard = ({ hostDetails, hostName }) => {
+  const {
+    compute_resource_id: computeResourceId,
+    compute_resource_name: computeResourceName,
+  } = hostDetails;
+  const virtUrl = foremanUrl(`/api/hosts/${hostName}/virtualization`);
+  const {
+    response: { vm, provider },
+    status,
+  } = useAPI('get', virtUrl);
+  if (vm === undefined) return null;
+
+  const components = {
+    vmware: VirtVmware,
+    libvirt: VirtLibvirt,
+    ovirt: VirtOvirt,
+    ec2: VirtEc2,
+    gce: VirtGce,
+    openstack: VirtOpenstack,
+  };
+  const VirtCardDetails = components[provider];
+
+  return (
+    <CardTemplate header={__('Virtualization')} expandable masonryLayout>
+      <SkeletonLoader
+        emptyState={<DefaultLoaderEmptyState />}
```

Comment:
empty state will never happen here because the skeletonLoader will always have children (`DescriptionList`)